src/simplify_dyn_ops.cpp

```diff
@@ -450,6 +451,83 @@ struct find_static_broadcast_for_dot
     }
 };
 
+/**
+ * Simplify onehot instructions with static shape `indices` input and
+ * a compile-time constant `depth` input.
+ * From:
+ * onehot(static_shape_arg, constant_arg, values)
+ * To:
+ * A = literal(shape = onehot_output_shape, value = 0)
+ * B = unsqueeze(literal(lens = indices_lens, strides = broadcasted scalar, value = 1), axis=onehot_axis)
+ * C = scatter(A, unsqueeze(indices, axis=onehot_axis), B)
+ * diff = on_value - off_value
+ * D = mul(diff, C);
+ * return = add(D, off_value);
+ *
+ * NOTE: It might be cleaner to use some form of `fill` instead of
+ * (on_value - off_value) * mask + off_value when we have `fill` working
+ * on the GPU.
+ */
+struct find_static_onehot
+{
+    auto matcher() const
+    {
+        return match::name("onehot")(match::arg(0)(match::static_shape()),
+                                     match::arg(1)(match::is_constant()),
+                                     match::arg(2)(match::static_shape()));
+    }
+
+    void apply(module& m, const match::matcher_result& mr) const
+    {
+        auto onehot_ins           = mr.result;
+        auto onehot_inputs        = onehot_ins->inputs();
+        auto onehot_op            = any_cast<op::onehot>(onehot_ins->get_operator());
+        auto indices_ins          = onehot_inputs[0];
+        shape indices_shape = indices_ins->get_shape();
+        auto depth_ins            = onehot_inputs[1];
+        auto values_ins           = onehot_inputs[2];
+        shape values_shape        = values_ins->get_shape();
+        std::vector<std::size_t> static_output_lens = indices_shape.lens();
+        auto normalized_axis =
```

Comment:
How does this handle axis = -1, the default, please. Also, the legit range of axis is [-r-1, r], r being the rank. Thanks.

Reply:
Adds `rank(indices) + 1` if negative. Handling the range of supported indices in the `scatter` operation.